internal/resources/metal/vlans/datasource_test.go

```diff
@@ -322,7 +277,7 @@ func TestMetalVlan_matchingVlan(t *testing.T) {
 }
 
 func testAccMetalDatasourceVlanCheckDestroyed(s *terraform.State) error {
-	client := testAccProvider.Meta().(*config.Config).Metal
+	client := acceptance.TestAccProvider.Meta().(*config.Config).Metal
 
 	for _, rs := range s.RootModule().Resources {
 		if rs.Type != "equinix_metal_vlan" {
```

Comment:
Looks good!  Guidance from @ocobles is to add an `_upgradeFromVersion` test as well so we can be more confident that we haven't introduced breaking changes by migrating to framework.  Could you add one test like that for the datasource and one for the resource.   Here's an example from Oscar's most recent framework migration PR: https://github.com/equinix/terraform-provider-equinix/pull/572/files#diff-dc80edad0c4ecf395554d4f118aa4dbb2f5963b52a7eb4ae8aa8e57184b1ee8eR128

Reply:
Yes, added upgrade tests as well. Passing locally, please approve